File: src/dotnet/EntityFX.NetBenchamarks.Core/Generic/MemoryBenchmark.cs
using System;
namespace EntityFX.NetBenchmark.Core.Generic
{
    public class MemoryBenchmark: MemoryBenchmarkBase<MemoryBenchmarkResult>
    {
        public MemoryBenchmark(bool printToConsole, IWriter writer)
            : base(printToConsole, writer)
        {
            localWriter.UseConsole = UseConsole;
        }

        public override MemoryBenchmarkResult BenchImplementation() 
        {
            return BenchRandomMemory();
        }

        public override BenchResult PopulateResult(BenchResult benchResult, MemoryBenchmarkResult result)
        {
            benchResult.Points = result.Average * Ratio;
            benchResult.Result = result.Average;
            benchResult.Units = "MB/s";
            benchResult.Output = result.Output;
            return benchResult;
        }

        public override void Warmup(double aspect)
        {
            localWriter.UseConsole = false;
            base.Warmup(aspect);
            localWriter.UseConsole = true;
        }

    }
}

File: src/python/entityfx/memory_benchmark.py

from entityfx.memory_benchmark_base import MemoryBenchmarkBase

class MemoryBenchmark(MemoryBenchmarkBase):
    
    def benchImplementation(self) :
        return self.benchRandomMemory()
    
    def populateResult(self, bench_result, result) :
        bench_result["Points"] = result["Average"] * self.ratio
        bench_result["Result"] = result["Average"]
        bench_result["Units"] = "MB/s"
        bench_result["Output"] = result["Output"]
        return bench_result